packages/manager/cypress/e2e/core/cloudpulse/aclp-support.spec.ts

```diff
@@ -0,0 +1,197 @@
+import { linodeFactory } from '@linode/utilities';
+import { regionFactory } from '@linode/utilities';
+import { mockGetCloudPulseDashboardByIdError } from 'support/intercepts/cloudpulse';
+import { mockAppendFeatureFlags } from 'support/intercepts/feature-flags';
+import { mockGetLinodeDetails } from 'support/intercepts/linodes';
+import { mockGetLinodeStats } from 'support/intercepts/linodes';
+import { mockGetUserPreferences } from 'support/intercepts/profile';
+import { mockGetRegions } from 'support/intercepts/regions';
+import { ui } from 'support/ui';
+import {
+  assertBetaMetricsNotAvailable,
+  generateMockLegacyStats,
+} from 'support/util/cloudpulse';
+import { randomLabel, randomNumber } from 'support/util/random';
+
+import type { Stats } from '@linode/api-v4';
+
+/**
+ * If feature flag and region supports aclp:
+ *  If user preference enables aclp, then UI displays beta metrics w/ option to switch to legacy metrics
+ *  If user preference disables aclp, then UI displays legacy metrics w/ option to switch to beta metrics
+ *
+ * If region does not support aclp, then the UI displays legacy metrics w/ no option to switch to beta metrics
+ */
+describe('ACLP Components UI varies according to ACLP support by region and user preference', function () {
+  beforeEach(function () {
+    mockAppendFeatureFlags({
+      aclpBetaServices: { metrics: true },
+    }).as('getFeatureFlags');
+  });
+  describe('toggle user preference when region supports aclp', function () {
+    beforeEach(function () {
+      const mockRegion = regionFactory.build({
+        capabilities: ['Managed Databases'],
+        monitors: {
+          metrics: ['Linodes'],
+        },
+      });
+      mockGetRegions([mockRegion]).as('getRegions');
+      const mockLinode = linodeFactory.build({
+        id: randomNumber(),
+        label: randomLabel(),
+        region: mockRegion.id,
+      });
+      cy.wrap(mockLinode.id).as('mockLinodeId');
+      mockGetLinodeDetails(mockLinode.id, mockLinode).as('getLinode');
+
+      // linodeDashboardId is hardcoded in LinodeMetrics.tsx to 2
+      // error param is not displayed
+      const mockDashboardId = 2;
+      cy.wrap(mockDashboardId).as('mockDashboardId');
+      const mockDashboardError = 'mock dashboard error occurred';
+      mockGetCloudPulseDashboardByIdError(
+        mockDashboardId,
+        mockDashboardError
+      ).as('getDashboardError');
+    });
+    // UI displays beta metrics, can switch to legacy view
+    it('user preference enables aclp', function () {
```

Comment:
style: we may want to use arrow functions everywhere for consistency. I think most, if not all of our tests use this syntax

```suggestion
    it('user preference enables aclp', () => {
```

Reply:
normally i do use arrow functions but cannot in this case bc i need to access cy.wrap(mockLinode).as('mockLinode'); from beforeEach within the 'it' fn w/ this.mockLinode.